x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -714,7 +717,9 @@ int getFreq(CacheFileRegion<KeyType> cacheFileRegion) {
 
     @Override
     public void close() {
-        sharedBytes.decRef();
+        if (closed.compareAndSet(false, true)) {
```

Comment:
Intentionally made `closed` a CAS, so we are able to call `close` multiple times from tests, but happy to preserve the current behavior where can call `close` only once.